torch/csrc/jit/interpreter.cpp

```diff
@@ -213,46 +485,83 @@ Operation getOperation(jit::Node *node) {
-    };
+    if(computes_on_variables) {
+      auto t = torch::autograd::make_variable(value->t(kvalue), false);
+      return [t](Stack & stack) {
+        stack.push_back(t);
+        return 0;
+      };
+    } else {
+      auto t = value->t(kvalue);
+      return [t](Stack & stack) {
+        stack.push_back(t);
+        return 0;
+      };
+    }
   IR_ELSEIF(Undefined)
-    return [](const list_of_retainable & inputs, list_of_retainable & outputs) {
-      outputs.push_back(toRetainableSteal(at::Tensor()));
+    return [](Stack & stack) {
+      stack.push_back(at::Tensor());
+      return 0;
     };
   IR_ELSEIF(ReplaceIfUndef)
-    return [](const list_of_retainable & inputs, list_of_retainable & outputs) {
-      auto result = inputs[0];
-      //TODO: refcounting stuff here is ugly but TensorTemporary is not
-      //present. Consider whether we
-      // 1. expose tensor temporary here
-      // 2. keep as is
-      // 3. remove all of this retainable stuff anyway since the new
-      // execution paths do not need handle types.
-      // Note that list_of_retainable is painful because it is yet another
-      // list of pointers that require needless copies.
-      if(result == at::UndefinedTensor::singleton()) {
-        result = inputs[1];
+    return [](Stack & stack) {
+      auto alternate = pop(stack);
+      auto result = pop(stack);
+      if(result.defined()) {
+        stack.push_back(std::move(result));
+      } else {
+        stack.push_back(std::move(alternate));
       }
-      result->retain();
-      outputs.push_back(result);
+      return 0;
     };
   IR_ELSEIF(GraphExecutor)
     GraphExecutor executor(value->g(kSubgraph));
-    return [=](const list_of_retainable & inputs, list_of_retainable & outputs) mutable {
+    auto num_inputs = value->inputs().size();
+    return [=](Stack& stack) mutable {
       autograd::profiler::RecordFunction record("GraphExecutor");
-      tensor_list tinputs = unsafeToTensorListShare(inputs);
+      auto inputs = last(stack, num_inputs);
+      variable_tensor_list tinputs(Stack(inputs.begin(), inputs.end()));
+      drop(stack, num_inputs);
+      //TODO: has graph executor work from a stack as well
       variable_tensor_list toutputs = executor.run(variable_tensor_list(std::move(tinputs)));
-      moveToListOfRetainables(std::move(toutputs), outputs);
+      stack.insert(stack.end(), toutputs.begin(), toutputs.end());
+      return 0;
+    };
+
+
+  // Load x, y
+  // loads values from registers onto the stack, the actual callback does
+  // nothing since the stack manipulation is already encoded in inst.inputs
+  // and inst.outputs
+  IR_ELSEIF(Load)
+    return [=](Stack& stack) {
+      return 0;
+    };
+
+  // x, y = Store
+  // stores values from stack into registers, the actual callback does
+  // nothing since the stack manipulation is already encoded in inst.inputs
+  // and inst.outputs
+  IR_ELSEIF(Store)
+    return [=](Stack& stack) {
+      return 0;
+    };
```

Comment:
Why do we even have those nodes then? Just to delimit stage ends in the node list? It would feel nicer to explicitly check for them and ignore when emitting instructions, instead of treating them as complete no-ops, no matter where they appear